users/models.py

```diff
@@ -1,3 +1,290 @@
+        """
+        Create and save a new superuser with the given email and password.
+
+        Args:
+            email (str): Superuser's email address.
+            password (str, optional): Superuser's password.
+            **extra_fields: Additional fields for the superuser.
+
+        Returns:
+            User: Newly created superuser instance.
+        """
+        extra_fields.setdefault('is_staff', True)
+        extra_fields.setdefault('is_superuser', True)
+        extra_fields.setdefault('is_active', True)
+        user = self.create_user(email, password, **extra_fields)
+        if 'Admin' not in self.ALLOWED_ROLES:
+            raise ValidationError(f"Cannot add role: Admin")
+        user.add_role('Admin')
+        logger.info(f"New superuser created: {email}")
+        return user
+
+
+class Role(models.Model):
+    """Model to represent user roles."""
+    name = models.CharField(max_length=50, unique=True)
+
+    def __str__(self):
+        return self.name
+
+
+class User(AbstractBaseUser, PermissionsMixin):
+    """
+    Custom user model extending Django's AbstractBaseUser and PermissionsMixin.
+
+    This model includes fields for user authentication, personal information,
+    and role management. It supports multiple roles for users, soft delete
+    functionality, and custom methods for role manipulation.
+    """
+    groups = models.ManyToManyField(Group, related_name='custom_user_groups')
+    user_permissions = models.ManyToManyField(Permission, related_name='custom_user_permissions')
+
+    ALLOWED_ROLES = ['Investor', 'Startup', 'Admin']
+
+    email = models.EmailField(max_length=255, db_index=True)
+    first_name = models.CharField(max_length=100)
+    last_name = models.CharField(max_length=100)
+    user_phone = models.CharField(
+        max_length=20,
+        validators=[RegexValidator(
+            regex=r'^\+?1?\d{9,15}$',
+            message="Phone number must be entered in the format: '+999999999'. Up to 15 digits allowed."
+        )]
+    )
+    roles = models.ManyToManyField(Role, related_name='users')
+    profile_picture = models.ImageField(upload_to='profile_pics/', blank=True, default='')
+    about_me = models.CharField(max_length=255, blank=True, default='')
+    is_active = models.BooleanField(default=True)
+    is_staff = models.BooleanField(default=False)
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+    is_soft_deleted = models.BooleanField(default=False)
+    deleted_at = models.DateTimeField(blank=True, null=True)
+    last_login = models.DateTimeField(null=True, blank=True)
+    original_data = models.JSONField(null=True, blank=True)
+
+    objects = CustomUserManager()
+
+    USERNAME_FIELD = 'email'
+    REQUIRED_FIELDS = ['first_name', 'last_name', 'user_phone']
+
+    def __str__(self):
+        """Return a string representation of the user."""
+        return self.email
```

Comment:
The `__str__` method could be enhanced to provide more user details for admin interfaces or debugging.